### ISSUE TITLE
Missing license

### ISSUE THREAD

**Issue by kattjevfel** [GTAmodding/re3#794]
I noticed this repo/project is lacking a license, could one please be added?

**Comment by hazelnot** [comment]
I'm strongly for a GPL3 license but also all I contributed to the project was like a bug report and a logo 😅

**Comment by aap** [comment]
We aren't sure if it even makes sense to give a license to such a project, or whether we're in the position to give it one. So the lack of a license is currently intentional. maybe we should add one just in case (it would probably be MIT), but somehow that feels weird to me.

**Comment by hazelnot** [comment]
Hmm, I mean I don't actually know whether this is a clean room reverse-engineering or not. If it is then I stand by my GPL3 suggestion, but if it isn't it's, uh, illegal and probably will get shut down by Rockstar and maybe shouldn't be publicly available on GitHub right now

**Comment by Calinou** [comment]
I suppose we could add a MIT license for all code that has been rewritten, and add a disclaimer about all other source files that haven't been rewritten significantly enough yet.

**Comment by kattjevfel** [comment]
Isn't all original GTA3 content only added by the end user when compiling? I thought the repo only contained the rewritten stuff.

**Comment by aap** [comment]
What? we have reversed engineered the complete game the same as other projects (super mario 64, diablo, ...) and all our code is in this repo.

**Comment by kattjevfel** [comment]
Great! That means there's no original content from rockstar then? What's the problem?

**Comment by kattjevfel** [comment]
What about the files in gamefiles/and utils/gxt/? Looking into some of these files have dates from 2000 and 2001, and I can only assume they're ripped from copies of the game, which would not be legal to distribute. Perhaps a better solution would be to fetch these files (from a user provided game copy) and patch any modifications done.

**Comment by x87** [comment]
Could be [Unlicense](https://choosealicense.com/licenses/unlicense)

**Comment by Clownacy** [comment]
I'm not a lawyer, but doesn't decompiled code count as a derivative work?

**Comment by Krutonium** [comment]
We did the same thing with RollerCoaster Tycoon 2 over at https://github.com/OpenRCT2/OpenRCT2/blob/develop/licence.txt - And we're using GPLv3. (RE'ing the original binary and re-writing it in C/C++). From a legal standpoint, not being lawyers, as far as we can tell, you can license it just about however you want.

**Comment by IacobusKopiirefuto** [comment]
There does not seem to be much of a progress on this issue.

If need more information about different licenses, GitHub has a site that can help you with that <https://choosealicense.com/>. This site has also info on what does in mean, when repository does not have a license <https://choosealicense.com/no-permission/>.

**Comment by Soundtoxin** [comment]
Just want no chime in to say this is All Right Reserved until you license it. Entirely proprietary. You say something like people lhould keep derivative works open source, but that's not possible. I suggest a copyleft license based on you wishing that.

**Comment by avoidr** [comment]
From the README:
>We don't feel like we're in a position to give this code a license.

Why? Why do you feel so? Please elaborate on this. It's impossible to find a solution if you don't tell us what you think the issue is.

>The code should only be used for educational, documentation and modding purposes.

Why?

>We do not encourage piracy or commercial use.

Why the disclaimer? You're not doing anything wrong (or are you?). And if you are, I don't think this disclaimer will help you.

>Please keep derivate work open source and give proper credit.

Impossible to satisfy, since your work is not open source in the first place, as @Soundtoxin noted. Based on that description, you'd probably like a copyleft license like the GPL.

>We aren't sure if it even makes sense to give a license to such a project

Why? Without any restrictions, the answer seems quite clear to me: It absolutely makes sense. Your "such a project" is hard work of hundreds or thousands of hours. You are the author and it absolutely makes sense to give this project a license.

A license is nothing more than an explicit written permission to your users. If you don't give a license, your work will default to proprietary "All Rights Reserved", due to the Berne Convention. Which means that your work cannot be packaged for repos of many distros.

>So the lack of a license is currently intentional. maybe we should add one just in case (it would probably be MIT), but somehow that feels weird to me.

Why??? What feels weird? Please answer us ;_;

**Comment by OmniBlade** [comment]
I would say that until a court/judge decides otherwise, the contributors to this project are the authors of the code here and can license it however they wish and indeed should do so.

**Comment by aap** [comment]
GTA is R*'s intellectual property and we cannot nor want to claim it is ours. We have reversed the code but it is still a derivate of R*'s work.
Reverse engineering is a grey zone and none of us is a lawyer. But would any license we might give it really hold up in court? I highly doubt it and if you can't enforce a license what's the point.

**Comment by zoj613** [comment]
If there is no license, then as I understand it, the work in this project cannot be used/modified/distributed by anyone since its technically not open source?

**Comment by aap** [comment]
You can find the source in this repository.

**Comment by zoj613** [comment]
> You can find the source in this repository.

Yes, but no one other than the creator of this repo has the right to use the source unless there is a clear license regarding usage terms.

**Comment by hazelnot** [comment]
Dude aap isn't gonna sue you

**Comment by aap** [comment]
Get a copyright lawyer to figure out the situation. otherwise i'm out of this discussion.

**Comment by zoj613** [comment]
I'm not concerned about being sued. I have no interest nor the capability to modify anything written in the source of this repo. I just figured i'd chime in since it is important to state usage terms if someone is sharing code publicly to be used by others.

**Comment by Soundtoxin** [comment]
> GTA is R*'s intellectual property

I recommend reading this.
https://www.gnu.org/philosophy/words-to-avoid.html#IntellectualProperty

It's a deceptive word trying to combine copyright, trademarks, and patents. Generally you only care about copyright with software. Assuming this is a cleanroom RE, you've done nothing wrong. Look at OpenMW for an example of RE'd code that's been freely licensed. You're not the first to go down this path, I'd say it's proven to be fine.

https://gitlab.com/OpenMW/openmw (GPLv3)

**Comment by OmniBlade** [comment]
On further reflection, I'm of the opinion that there is a license after reading the readme, albeit a very vague and poorly defined one. Terms of use are educational, documentation and modding purposes only, piracy and commercial useage are allowed but not encouraged and derivitive works must give credit and also be open source. Despite not feeling in the position to give it one it seems the authors have done so none the less.

**Comment by aap** [comment]
We made our intentions clear. It's not supposed to be a license in any legal sense.

**Comment by OmniBlade** [comment]
Saying you don't feel that you can, doesn't mean that the subsequent lines don't form one. You state terms of use of the code which is all a license really is.

**Comment by avoidr** [comment]
>GTA is R*'s intellectual property and we cannot nor want to claim it is ours.

Yes, GTA is a work of Rockstar. Rockstar is the copyright holder of GTA.
re3 is your work. You are the copyright holder of re3.

The copyright holder has the power to grant rights (add license) to the users of his work. If the copyright holder does not grant others any rights, he *still has the copyrights to his work*. License != copyright. Not granting others any rights does not in any way protect you legally from infringing on other people's copyrights. Much less do disclaimers like yours, or the popular "No copyright infringement intended", give you any kind of legal immunity.

But not granting others any rights, others may not use your work. They may not share it or change it, for example. Only you may. Because you have not granted anyone else any rights.

>We have reversed the code but it is still a derivate of R*'s work.

re3 is no more derivate of Rockstar's work, than LibreOffice is a derivate of Microsoft's work.

>Reverse engineering is a grey zone and none of us is a lawyer.

And yet you already published your work. That's potential copyright infringement, whether you add a license or not.

>But would any license we might give it really hold up in court?

Yeah, that's the point of copyright. Whether *you* are infringing on Rockstar's copyright is a completely separate issue, though.

>I highly doubt it and if you can't enforce a license what's the point.

Then I suggest you waive all of your rights and put re3 into the public domain.

>You can find the source in this repository.

That's not what "open source" means.

Definition of "Open Source": https://opensource.org/osd
Definition of "Free Software": https://www.gnu.org/philosophy/free-sw.en.html

(There is no shame in not knowing that those terms are defined. I myself haven't known this for a long time, simply because those terms are just *too* intuitive.)

>Get a copyright lawyer to figure out the situation. otherwise i'm out of this discussion.

No no, that was *your* homework to figure out the situation, beforehand. So if you want to be sure that you're not infringing on Rockstar's copyright, maybe you should pull down the repo until you do figure it out. But do know that whether you put a license on re3 or not is completely unrelated to whether re3 itself infringes on Rockstar's copyright or not.

If you're really serious about clearing up any doubts, I'm sure you can contact the FSF (Free Software Foundation), they will be able to help you. Software freedom is their mission.
https://www.fsf.org/about/contact/

>We made our intentions clear. It's not supposed to be a license in any legal sense.

Then unfortunately that means that re3 is non-free, not open source, proprietary software, and we're back to square one; the situation of not having a libre implementation of a proprietary software. In the free-software/open-source world, the *entire point* of reverse engineering, is to have a libre implementation so that the user is not restricted, and then to *retain* the freedoms by putting a "copyleft" license on the work.

I try so hard trying to convince you, because I realize that we're just **so close** to a **huge** contribution to GTA gaming. Having a libre engine for GTA III *and* Vice City and being able to play on *any* platform, fix bugs and implement new features, and being able to package it and just install it from my distro's repo, that's the fucking **dream**. I don't know if you realize how big of a deal this is. It would be a real shame to let all this work go unused.

**Comment by vilhelmgray** [comment]
For those wondering what the current license text is according to the README file:
```
We don't feel like we're in a position to give this code a license.
The code should only be used for educational, documentation and modding purposes.
We do not encourage piracy or commercial use.
Please keep derivate work open source and give proper credit.
```

It looks like the current license therefore has 3 requirements:
* Permitted use of code is limited to the only the following three activities: education, documentation, and modding.
* Derivative works must be open source.
* Attribution must be given.

It seems unintentional, but unfortunately the "permitted use" clause here significantly limits what users can do with this source code (limited to only three specific activities). It would be prudent to adopt to use a well-know license to avoid these sort of unintended pitfalls.

If you don't want to adopt a specific license then at least release it to the public domain as suggested earlier in this thread in order to prevent the confusion we're witnessing here in this Github PR.

**Comment by hristoast** [comment]
> > GTA is R*'s intellectual property
> 
> I recommend reading this.
> https://www.gnu.org/philosophy/words-to-avoid.html#IntellectualProperty
> 
> It's a deceptive word trying to combine copyright, trademarks, and patents. Generally you only care about copyright with software. Assuming this is a cleanroom RE, you've done nothing wrong. Look at OpenMW for an example of RE'd code that's been freely licensed. You're not the first to go down this path, I'd say it's proven to be fine.
> 
> https://gitlab.com/OpenMW/openmw (GPLv3)

I want to clarify: OpenMW **is not RE'd code**. It is an original re-implementation.

**Comment by Krutonium** [comment]
OpenRCT2, however, is: https://github.com/OpenRCT2/OpenRCT2/ https://github.com/OpenRCT2/OpenRCT2/blob/develop/licence.txt
